### PR TITLE
XRDDEV-669

### PR DESCRIPTION
**Pull request by carohauta**
## Description

Refactor all `group` related stuff to `local-group` / `global-group`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have followed the agreed [Version Control Practices](https://confluence.niis.org/pages/viewpage.action?spaceKey=XRDDEV&title=Version+Control+Practices)
- [x] My changes generate no new warnings or errors (e.g. Javascript console, Java stdout)
- [x] I have made corresponding changes to the documentation
- [x] The new code has sufficient test coverage
- [x] The build, unit and integration tests pass
- [x] There is a link to a successful Jenkins build
- [x] No new npm audit issues, or new issues have been added to [Front-end build process](https://confluence.niis.org/display/XRDDEV/Front-end+build+process) tracking table and accepted
- [x] New backlog items that have been created (such as "not implementing this acceptance criteria now") are mentioned + linked in Jira task comments
- [x] All task outputs (PR, documentation, UI design, etc) is listed in a Jira comment so that it is easy for reviewer to check
- [x] Deviations from acceptance criteria listed in comments (also if "this criteria was removed")

### CODE REVIEW

File: src/proxy-ui-api/src/main/resources/openapi-definition.yaml [hunk 1]
```diff
@@ -1184,7 +1184,99 @@ paths:
           description: an existing item already exists
         '500':
           description: internal server error
-  /groups/{group_id}:
+  /clients/{id}/subjects:
```
Comment (jansu76) [inline src/proxy-ui-api/src/main/resources/openapi-definition.yaml]: Is this from XRDDEV-664? Is it in this PR for a reason (probably), or should we wait to merge 664 first?

File: src/proxy-ui-api/src/main/resources/openapi-definition.yaml [hunk 2]
```diff
@@ -650,7 +650,7 @@ paths:
           description: resource requested does not exists
         '500':
           description: internal server error
-  /clients/{id}/groups:
+  /clients/{id}/local-groups:
```
Comment (jansu76) [inline src/proxy-ui-api/src/main/resources/openapi-definition.yaml]: Probably should update API version to 1.0.11?

File: src/proxy-ui-api/src/main/resources/openapi-definition.yaml (hunk 1 above)
Comment (carohauta) [reply src/proxy-ui-api/src/main/resources/openapi-definition.yaml]: No it's not here on purpose (should be in branch 664 only). Reverted! 😅

File: src/proxy-ui-api/src/main/resources/openapi-definition.yaml (hunk 2 above)
Comment (carohauta) [reply src/proxy-ui-api/src/main/resources/openapi-definition.yaml]: Updated the version

**Pull request merged**